app/src/main/java/com/firebase/uidemo/database/ChatActivity.java

```diff
@@ -182,4 +177,12 @@ private void updateUI() {
         mSendButton.setEnabled(isSignedIn());
         mMessageEdit.setEnabled(isSignedIn());
     }
+
+    // TODO remove once arch components are merged into support lib
```

Comment:
Yeah, you can see the [documentation for `LifecycleActivity`](https://developer.android.com/reference/android/arch/lifecycle/LifecycleActivity.html). Basically, Fragments and activities will be lifecycle owners once AC hits stable.

PS: I didn't extend `LifecycleActivity` because it extends `FragmentActivity` instead of `AppCompatActivity`.